mobx_codegen/test/data/valid_import_prefixed_output.dart

```diff
@@ -138,8 +143,8 @@ mixin _$User<T extends io.Process> on UserBase<T>, Store {
     });
   }
 
-  final _$friendWithExplicitTypeArgumentAtom =
-  Atom(name: 'UserBase.friendWithExplicitTypeArgument');
+  late final _$friendWithExplicitTypeArgumentAtom =
```

Comment:
I am a bit worried about the cost of "late final", since it imposes runtime checks

Reply:
@fzyzcjy Yes I agree.

Another way I've considered is:
```dart

/// A base class to use custom context.
class Foo<ReactiveContext> {
  final ReactiveContext context;

  Foo(this.context);
}

class CustomContextStore = _CustomContextStore with _$CustomContextStore;

final ReactiveContext customContext = ReactiveContext();

abstract class _CustomContextStore extends Foo with Store {
  @observable
  late String name;

  _CustomContextStore(context) : super(context);

}

void main() {
  test('use custom context', () {
    final store = CustomContextStore(customContext);

    expect(store._$nameAtom.context, equals(customContext));
  });
}
```
However, there is other downside to using `extends`. Is there any other good options?

Reply:
Agree. Have not come up with other approaches yet